Keen/Event.cs

```diff
@@ -2,48 +2,86 @@
 using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.IO;
 using System.Linq;
-using System.Net.Http;
-using System.Text;
 using System.Threading.Tasks;
 
+
 namespace Keen.Core
 {
     /// <summary>
     /// Event implements the IEvent interface which represents the Keen.IO Event API methods.
     /// </summary>
     internal class Event : IEvent
     {
-        private IProjectSettings _prjSettings;
-        private string _serverUrl;
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _eventsRelativeUrl;
+        private readonly string _readKey;
+        private readonly string _writeKey;
+
+
+        internal Event(IProjectSettings prjSettings,
+                       IKeenHttpClientProvider keenHttpClientProvider)
+        {
+            if (null == prjSettings)
```

Comment:
There are strange thinks like Queries.cs housing the generic "call-all" api shim, whereas that should be somewhere shared by all of the various top-level APIs. This will require a bigger refactor to correct, for sure.